model/group_request.go

```diff
@@ -111,8 +95,14 @@ func (p *PatchGroupRequest) Apply(group *Group) bool {
 		group.MaxRolling = *p.MaxRolling
 	}
 	if p.MattermostEnv != nil {
-		applied = true
-		group.MattermostEnv = p.MattermostEnv
+		if group.MattermostEnv != nil {
+			if group.MattermostEnv.ClearOrPatch(&p.MattermostEnv) {
```

Comment:
0/5 could remove the conditional and just say `applied = group.MattermostEnv...`

Reply:
Oh, just remembered that I can't actually do that as applied could be `true` for the other patches and then this would set it back to `false` incorrectly.